src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,300 @@
+---
+date: "July 27 2023"
+description: "Why it's not practically feasible -or desirable- for a programming language like TypeScript to use strict semantic versioning for its releases."
+meta: ECMAScript, compiler options target, TSConfig
+---
+
+import { CaptionedImage } from "@site/src/theme/CaptionedImage";
+
+# Why TypeScript Doesn't Follow Strict Semantic Versioning
+
+Most projects in the JavaScript/TypeScript ecosystem release new versions with numbers respecting to _**[semantic versioning](https://semver.org)**_, or _semver_ for short.
+TypeScript is notable for not following a traditional interpretation of semver for its releases.
+This article will dig into:
+
+1. What semver is and why it's useful for many packages
+2. Why following a strict interpretation of semver would be impractical for TypeScript
+3. How TypeScript's releases are versioned to an interpretation of semver that makes sense for it
+
+While TypeScript's diverging from a common community specification can be irksome for developers, there are real reasons why it chose to do so.
+
+<!-- truncate -->
+
+## Context: Semantic Versioning
+
+Semver is a specification that describes how to increase a package's version numbers upon each new release.
+Specifically, it dictates that versions adhere to a _Major.Minor.Patch_ format:
+
+- _Major_: increases when you make incompatible API changes
+- _Minor_: increases when you add functionality in a backward compatible manner
+- _Patch_: increases when you make backward compatible bug fixes
+
+For example, given a package at version `1.2.3`, it might increase its version to:
```

Comment:
[nitpick] I'm having trouble understanding this sentence. Maybe you can remove "For example"? 
```suggestion
For example, given a package at version `1.2.3`, it might increase its version to:
```

Reply:
That does, thanks. My instinct in cases where the sentence flow is unclear is to try the split the sentence in two. I have a tendency to write long sentences.

Reply:
```suggestion
For example, suppose a package is at version `1.2.3`.
It might increase its version to:
```